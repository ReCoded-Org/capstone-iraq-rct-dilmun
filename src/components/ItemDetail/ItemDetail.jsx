import React from 'react'
import PropTypes from 'prop-types'

import { useTranslation } from 'react-i18next'

export default function ProductDetail({ title, price, description }) {
    const { t } = useTranslation()
    return (
        <>
            <b className="block m-1">{title}</b>
            <p className="bg-yellow p-1 m-1 text-black inline-block">
                {t('proudctDetail.price')}: $ {price}
            </p>
            <b className="block m-1 mt-3"> {t('proudctDetail.description')}</b>
            <p className=" m-1 text-black ">{description}</p>
        </>
    )
}

ProductDetail.propTypes = {
    title: PropTypes.string,
    price: PropTypes.number,
    description: PropTypes.string,
}

ProductDetail.defaultProps = {
    title: '',
    price: '',
    description: '',
}
