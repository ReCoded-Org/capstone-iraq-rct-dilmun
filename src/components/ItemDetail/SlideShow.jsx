import React from 'react'
import { Carousel } from 'react-responsive-carousel'
import 'react-responsive-carousel/lib/styles/carousel.min.css'

import Img1 from '../../assets/classy.jpg'
import Img2 from '../../assets/profile_bg.png'

export default function SlideShow() {
    const myImgs = [Img1, Img2]

    return (
        <div>
            <div className="grid gric grid-cols-1 lg:grid-cols-12  ">
                <Carousel
                    showArrows
                    showThumbs
                    className="lg:col-span-8 lg:col-start-2  "
                >
                    {myImgs.map((img) => {
                        return (
                            <div className="border-2 border-grey rounded-xl  overflow-hidden p-1  ">
                                <img
                                    alt=""
                                    src={img}
                                    className="  object-contain max-h-96 w-full "
                                />
                            </div>
                        )
                    })}
                </Carousel>
            </div>
        </div>
    )
}
