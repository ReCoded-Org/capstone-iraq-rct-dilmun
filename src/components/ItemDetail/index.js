export { default } from './ItemDetail'
export { default as SlideShow } from './SlideShow'
