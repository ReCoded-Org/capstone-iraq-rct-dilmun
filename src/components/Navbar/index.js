export { default } from './NavBar'
export { default as SecondaryNavbar } from './SecondaryNavbar'
