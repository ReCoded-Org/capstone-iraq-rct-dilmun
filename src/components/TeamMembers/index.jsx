export { default } from './TeamMembers'
export { default as AboutUs } from './AboutUs'
