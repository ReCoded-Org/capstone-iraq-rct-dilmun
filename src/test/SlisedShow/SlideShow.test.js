import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import { SlideShow } from '../../components/ItemDetail'

it('test: SlideShow without any props', () => {
  const tree = renderer
    .create(
      <Router>
        <SlideShow />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
