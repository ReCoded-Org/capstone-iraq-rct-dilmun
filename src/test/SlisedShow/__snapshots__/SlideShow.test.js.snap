// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test: SlideShow without any props 1`] = `
<div>
  <div
    className="grid gric grid-cols-1 lg:grid-cols-12  "
  >
    <div
      className="carousel-root lg:col-span-8 lg:col-start-2  "
      tabIndex={0}
    >
      <div
        className="carousel carousel-slider"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <ul
          className="control-dots"
        >
          <li
            aria-label="slide item 1"
            className="dot selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={0}
          />
          <li
            aria-label="slide item 2"
            className="dot"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            value={1}
          />
        </ul>
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev control-disabled"
          onClick={[Function]}
          type="button"
        />
        <div
          className="slider-wrapper axis-horizontal"
          style={Object {}}
        >
          <ul
            className="slider animated"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransform": "translate3d(0,0,0)",
                "MozTransitionDuration": "350ms",
                "MsTransform": "translate3d(0,0,0)",
                "MsTransitionDuration": "350ms",
                "OTransform": "translate3d(0,0,0)",
                "OTransitionDuration": "350ms",
                "WebkitTransform": "translate3d(0,0,0)",
                "WebkitTransitionDuration": "350ms",
                "msTransform": "translate3d(0,0,0)",
                "msTransitionDuration": "350ms",
                "transform": "translate3d(0,0,0)",
                "transitionDuration": "350ms",
              }
            }
          >
            <li
              className="slide selected"
              onClick={[Function]}
            >
              <div
                className="border-2 border-grey rounded-xl  overflow-hidden p-1  "
              >
                <img
                  alt=""
                  className="  object-contain max-h-96 w-full "
                  src="classy.jpg"
                />
              </div>
            </li>
            <li
              className="slide"
              onClick={[Function]}
            >
              <div
                className="border-2 border-grey rounded-xl  overflow-hidden p-1  "
              >
                <img
                  alt=""
                  className="  object-contain max-h-96 w-full "
                  src="profile_bg.png"
                />
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next"
          onClick={[Function]}
          type="button"
        />
        <p
          className="carousel-status"
        >
          1 of 2
        </p>
      </div>
      <div
        className="carousel"
      >
        <div
          className="thumbs-wrapper axis-vertical"
        >
          <button
            aria-label="previous slide / item"
            className="control-arrow control-prev control-disabled"
            onClick={[Function]}
            type="button"
          />
          <ul
            className="thumbs animated"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransform": "translate3d(0,0,0)",
                "MozTransitionDuration": "350ms",
                "MsTransform": "translate3d(0,0,0)",
                "MsTransitionDuration": "350ms",
                "OTransform": "translate3d(0,0,0)",
                "OTransitionDuration": "350ms",
                "WebkitTransform": "translate3d(0,0,0)",
                "WebkitTransitionDuration": "350ms",
                "msTransform": "translate3d(0,0,0)",
                "msTransitionDuration": "350ms",
                "transform": "translate3d(0,0,0)",
                "transitionDuration": "350ms",
              }
            }
          >
            <li
              aria-label="slide item 1"
              className="thumb selected"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "width": 80,
                }
              }
              tabIndex={0}
            >
              <img
                alt=""
                className="  object-contain max-h-96 w-full "
                src="classy.jpg"
              />
            </li>
            <li
              aria-label="slide item 2"
              className="thumb"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "width": 80,
                }
              }
              tabIndex={0}
            >
              <img
                alt=""
                className="  object-contain max-h-96 w-full "
                src="profile_bg.png"
              />
            </li>
          </ul>
          <button
            aria-label="next slide / item"
            className="control-arrow control-next control-disabled"
            onClick={[Function]}
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
